shogun-lib/src/main/java/de/terrestris/shogun/lib/listener/LoginListener.java

```diff
@@ -47,21 +58,28 @@ public synchronized void onApplicationEvent(InteractiveAuthenticationSuccessEven
 
         String keycloakUserId = SecurityContextUtil.getKeycloakUserIdFromAuthentication(authentication);
 
-        // add missing user to shogun db
+        // Add missing user to shogun db
         Optional<User> userOptional = userRepository.findByKeycloakId(keycloakUserId);
         User user = userOptional.orElse(null);
         if (user == null) {
             user = new User(keycloakUserId, null, null, null);
             userRepository.save(user);
+
+            // Add admin instance permissions for the user.
+            userInstancePermissionService.setPermission(user, user, PermissionCollectionType.ADMIN);
```

Comment:
Looks like every user becomes an admin but probably i just don't get what happens here...

Reply:
Please double check!

Reply:
As stated above `Newly created users will get the Permssion.ADMIN on their own instances.` So the user is able to administer himself (update, delete)

Reply:
No, actually this saves the `ADMIN` permission to the instance of the user itself - it is not connected to any other user or role. Without this line, the user wouldn't be allowed to execute any instance related read, update or delete operation.